### PR TITLE
Better types

### LINKED ISSUE #1: Simpler Types
Right now it seems like there can only be up to 4 arguments, and the type definitions look very messy:

```ts
export type EmitSpec =
  [0] |
  [0, () => void] |
  [0, () => void, any] |
  [1] |
  [1, (a: any) => void] |
  [1, (a: any) => void, any] |
  [2] |
  [2, (a: any, b: any) => void] |
  [2, (a: any, b: any) => void, any] |
  [3] |
  [3, (a: any, b: any, c: any) => void] |
  [3, (a: any, b: any, c: any) => void, any] |
  [4] |
  [4, (a: any, b: any, c: any, d: any) => void] |
  [4, (a: any, b: any, c: any, d: any) => void, any] |
  [-1] |
  [-1, (...args: any[]) => void] |
  [-1, (...args: any[]) => void, any];
```

- [x] I can try making it so that the types are simpler and more flexible with the arguments and make a PR for it

### PR DESCRIPTION
**Pull request by ChocolateLoverRaj**
Closes #1

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -18,175 +20,29 @@ export class EventEmitter<E extends EmitSpec = [-1]> {
...
   removeAllListeners(): this;
 }
 
-/**
- * The observable half of an event emitter.
- * Omits emit APIs so you can only bind to events, not generate them.
- */
-export type Observable<E extends EmitSpec = [-1]> = {
-  [K in 'removeListener' | 'off' | 'once' | 'on' | 'addListener']: EventEmitter<EmitSpec>[K];
-}
```
Comment (cspotcode) [inline index.d.ts:52]: Why was this removed?  I'm sure you have a good reason for removing it, but I haven't looked at this code in a long time so I'm confused.

File: index.d.ts [hunk 2]
```diff
@@ -1,14 +1,16 @@
 export {EventEmitter as default};
 
+export type Listener<Params extends unknown[] = []> = (...params: Params) => void
+
```
Comment (cspotcode) [inline index.d.ts:4]: This type is the replacement for `ListenerFn`, right?

**Review by cspotcode — COMMENTED** [review]
Looks good, thank you.  I have a couple questions, but I don't think the code needs any changes.

File: index.d.ts (hunk 2 above)
Comment (ChocolateLoverRaj) [reply index.d.ts:4]: Yes

File: index.d.ts (hunk 1 above)
Comment (ChocolateLoverRaj) [reply index.d.ts:52]: I don't know why that line was there, so I removed it. There might be some code that references it, and if there is I'll add it back, but I didn't see anywhere that is was referenced, or how someone using the package would use `Observable`.

Comment (cspotcode) [reply index.d.ts:52]: I think it was meant for users who want to expose only the listening API, not the emitting API, of an `EventEmitter`.  Usage might look like this:

```
interface ThingThatEmitsEvents {
  everySecond: Observable<>;
  // ....
}
function createThing(): ThingThatEmitsEvents {
  const everySecond = new EventEmitter();
  setInterval(() => onSecond.emit(), 1000);
  // 
  return {
    everySecond,
    // ...
  };
}
const thing = createThing();
thing.everySecond.on() // can register listeners
thing.everySecond.emit() // ERROR cannot emit; only gets the Observable API
```

**Comment by cspotcode** [comment]
I re-added the `Observable` type; let me know what you think.  If it looks good to you, I'll merge and publish a new version.

**Comment by ChocolateLoverRaj** [comment]
@cspotcode, Sorry I didn't see your comment when you re-added the `Observable`, so I also re-added it and accidentally force pushed an amended commit with the observable.

**Comment by cspotcode** [comment]
No problem, as long as it's there.

**Comment by cspotcode** [comment]
Thanks, this has been published in 0.3.0: https://www.npmjs.com/package/single-event-emitter
